feature/604/1

AnimeOneLove2190 opened the pull request:
Add Movie and Genre Interfaces,
Add Movie and Genre DataAccess,
Add Movie and Genre Create Services,
Add Movie and Genre DTO Create Models,
Add MovieController

AnimeOneLove2190 committed:
feature/604/1
Add Movie and Genre Interfaces,
Add Movie and Genre DataAccess,
Add Movie and Genre Create Services,
Add Movie and Genre DTO Create Models,
Add MovieController

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/GenreService.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace CinemaTicket.BusinessLogicServices
+{
+    public class GenreService
```
Связать с IGenreService

artemlipinskiy replied:
И для всех сервисов и датааксессов задать зависимость в методе Startup ConfigureServices

AnimeOneLove2190 replied:
Как задать зависимость внутри метода ConfigureService?

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/GenreService.cs:
```diff
@@ -0,0 +1,44 @@
+        {
+            this.genreDataAccess = genreDataAccess;
+        }
+        public async Task CreateAsync(GenreCreate genreCreate) // Эта мразь не может увидеть GenreCreate, который находится в проекте DTO, поэтому пришлось засунуть его копию в проект с сервисами
```
Не вижу проблемы, используется же GenreCreate с DTO

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/MovieService.cs:
```diff
@@ -0,0 +1,73 @@
+
+namespace CinemaTicket.BusinessLogicServices
+{
+    class MovieService
```
Связать с IMovieService

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/MovieService.cs:
```diff
@@ -0,0 +1,73 @@
+            {
+                throw new Exception("CreateMovie: Movie Duration field must not be empty or contain a negative value");
+            }
+            var movieFromDB = movieDataAccess.GetMovieAsync(movieName);
```
await

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/MovieService.cs:
```diff
@@ -0,0 +1,73 @@
+            var movieFromDB = movieDataAccess.GetMovieAsync(movieName);
+            if (movieFromDB != null)
+            {
+                if (movieFromDB.Result.Name.ToLower() == movieName.ToLower() && movieFromDB.Result.Description.ToLower() == movieDescription.ToLower())
```
movieFromDB.Name.ToLower()

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/MovieService.cs:
```diff
@@ -0,0 +1,73 @@
+                ModifiedOn = DateTime.Now
+            };
+            await movieDataAccess.CreateAsync(movie);
+            var genresFromDB = genreDataAccess.GetGenreListAsync().Result.Select(x => x.Name.ToLower()).ToList();
```
await

artemlipinskiy commented on CinemaTicket.BusinessLogic.Interfaces/IGenreDataAccess.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace CinemaTicket.BusinessLogic.Interfaces
+{
+    public interface IGenreDataAccess
```
Перенести в CinemaTicket.DataAccess.Interfaces

artemlipinskiy commented on CinemaTicket.BusinessLogic.Interfaces/IMovieDataAccess.cs:
```diff
@@ -0,0 +1,19 @@
+
+namespace CinemaTicket.BusinessLogic.Interfaces
+{
+    public interface IMovieDataAccess
```
Перенести в CinemaTicket.DataAccess.Interfaces

artemlipinskiy commented on CinemaTicket.DataTransferObjects/Genres/GenreCreate.cs:
```diff
@@ -0,0 +1,14 @@
+    {
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public DateTime CreatedOn { get; set; }
```
Думаю, что модель создания не содержит CreatedOn и ModifiedOn

artemlipinskiy commented on CinemaTicket.DataTransferObjects/Movies/MovieCreate.cs:
```diff
@@ -0,0 +1,13 @@
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public int Duration { get; set; }
+        public DateTime CreatedOn { get; set; }
```
Думаю, что модель создания не содержит CreatedOn и ModifiedOn

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/MovieService.cs:
```diff
@@ -0,0 +1,73 @@
+            if (needAddGenres != null && needAddGenres.Count > 0)
+            {
+                var genres = needAddGenres.Select(x => new Genre
+                {
```
Нужно добавить в созданные жанры фильм

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/MovieService.cs:
```diff
@@ -0,0 +1,73 @@
+            {
+                if(genresFromDB.Contains(genreNames[i].ToLower()))
+                {
+                    continue;
```
Жанры, которые уже существуют, стоит добавить в фильм

artemlipinskiy commented on CinemaTicket.WebApi/Controllers/MovieController.cs:
```diff
@@ -0,0 +1,27 @@
+        }
+        [HttpPost]
+        [Route("AddMovie")]
+        public async Task CreateMovieAsync(string movieName, string movieDescription, int movieDuration, List<string> genreNames)
```
Все параметры можно объединить в модель CinemaCreate

artemlipinskiy commented on CinemaTicket.DataAccess.Interfaces/MovieDataAccess.cs:
```diff
@@ -2,7 +2,7 @@
 namespace CinemaTicket.DataAccess.Interfaces
 {
-    public class Class1
+    public interface IMovieDataAccess
```
Сам файл тоже переименовать в IMovieDataAccess

artemlipinskiy commented on CinemaTicket.DataAccess/MovieDataAccess.cs:
```diff
@@ -0,0 +1,49 @@
+
+namespace CinemaTicket.DataAccess
+{
+    class MovieDataAccess
```
Связать с IMovieDataAccess

artemlipinskiy commented on CinemaTicket.DataAccess/MovieDataAccess.cs:
```diff
@@ -0,0 +1,49 @@
+        public async Task UpdateMovieAsync(Movie movie)
+        {
+            movie.ModifiedOn = DateTime.Now;
+            cinemaManagerContext.Entry(movie).State = EntityState.Modified; // Что это за команда?
```
Помогает решить проблему, если сущность получена как AsNoTracking, аналогично для Deleted

AnimeOneLove2190 committed:
feature/604/2
The IMovieDataAccess and IGenreDataAccess interfaces have been moved to the DataAccess.Interfaces folder
A connection has been established between class objects and their interfaces
The input parameters in the MovieService method have been changed, and other changes have also been made there

AnimeOneLove2190 committed:
feature/604/3
Additional edits have been made, but the program does not start

AnimeOneLove2190 committed:
feature/604/4
Additional edits have been made
Test run was successful
